src/main/kotlin/gg/essential/elementa/components/ScrollComponent.kt

```diff
@@ -95,31 +143,45 @@ class ScrollComponent @JvmOverloads constructor(
     val verticalOverhang: Float
         get() = max(0f, calculateActualHeight() - getHeight())
 
+
+
+    val mouseScrollLambda: UIComponent.(UIScrollEvent) -> Unit = {
+        if (Window.of(this).version >= ElementaVersion.v5) {
+            // new behavior
+            val scrollDirection = if (!UKeyboard.isShiftKeyDown()) primaryScrollDirection else secondaryScrollDirection
+            scrollDirection?.let { direction ->
+                if (!onScroll(it.delta.toFloat(), isHorizontal = direction == Direction.Horizontal) && passthroughScroll) {
+                    getNextHighestScrollComponent()?.fireScrollEvent(it)
+                }
+            }
+
+            it.stopPropagation()
```

Comment:
Probably shouldn't stop propagation if we didn't handle it. Unless there's a reason we need to that I don't see.

On the contrary, consider e.g. a vertical-only scroller inside a prefer-vertical scroller. If the user shift+scrolls, the inner one won't handle it but will cancel the event, even though the expected behavior would probably be for the outer one to handle it.